Generated Hubspot CRM object products connector for Ballerina (#2)
ballerina-platform/module-ballerinax-hubspot.crm.object.products: README.md

     --url https://api.hubapi.com/oauth/v1/token \
     --header 'content-type: application/x-www-form-urlencoded' \
     --data 'grant_type=authorization_code&code=<CODE>&redirect_uri=<YOUR_REDIRECT_URI>&client_id=<YOUR_CLIENT_ID>&client_secret=<YOUR_CLIENT_SECRET>'
     ```

   - Windows

     ```bash
     curl --request POST ^
     --url https://api.hubapi.com/oauth/v1/token ^
     --header 'content-type: application/x-www-form-urlencoded' ^
     --data 'grant_type=authorization_code&code=<CODE>&redirect_uri=<YOUR_REDIRECT_URI>&client_id=<YOUR_CLIENT_ID>&client_secret=<YOUR_CLIENT_SECRET>'
     ```

   This command will return the access token necessary for API calls.

   ```json
   {
     "token_type": "bearer",
     "refresh_token": "<Refresh Token>",
     "access_token": "<Access Token>",
     "expires_in": 1800
   }
   ```

5. Store the access token securely for use in your application.

## Quickstart

To use the **HubSpot CRM Object Products** connector in your Ballerina application, update the `.bal` file as follows:

### Step 1: Import the module

Import the `hubspot.crm.obj.products` module and `oauth2` module.

```ballerina
import ballerina/oauth2;

> mport ballerinax/hubspot.crm.obj.products as hsproducts;

Reviewer: ```suggestion
import ballerinax/hubspot.crm.obj.products as hsproducts;
```

Author: Done:[3c1fff3](https://github.com/ballerina-platform/module-ballerinax-hubspot.crm.object.products/pull/2/commits/3c1fff3f902786febfcc410ab18c4db3e671cb9d)

> ## Overview

Reviewer: Can you restructure the `Overview` by referring to the following ?

https://github.com/ballerina-platform/module-ballerinax-hubspot.crm.object.deals/blob/d9c9c839279da32f429477537ccd250b399f02af/README.md#overview

Author: Done:[cf70c14](https://github.com/ballerina-platform/module-ballerinax-hubspot.crm.object.products/pull/2/commits/cf70c14111bd6f1e2578ae1728d8b47516dbafec)

Reviewer: I think this has not been done properly. Can you check again ?

Author: Done: [14bf194](https://github.com/ballerina-platform/module-ballerinax-hubspot.crm.object.products/pull/2/commits/14bf194d4daefa0da532db9ebe84ba440ea7b024)

     --url https://api.hubapi.com/oauth/v1/token \
     --header 'content-type: application/x-www-form-urlencoded' \
     --data 'grant_type=authorization_code&code=<CODE>&redirect_uri=<YOUR_REDIRECT_URI>&client_id=<YOUR_CLIENT_ID>&client_secret=<YOUR_CLIENT_SECRET>'
     ```

   - Windows

     ```bash
     curl --request POST ^
     --url https://api.hubapi.com/oauth/v1/token ^
     --header 'content-type: application/x-www-form-urlencoded' ^
     --data 'grant_type=authorization_code&code=<CODE>&redirect_uri=<YOUR_REDIRECT_URI>&client_id=<YOUR_CLIENT_ID>&client_secret=<YOUR_CLIENT_SECRET>'
     ```

   This command will return the access token necessary for API calls.

   ```json
   {
     "token_type": "bearer",
     "refresh_token": "<Refresh Token>",
     "access_token": "<Access Token>",
     "expires_in": 1800
   }
   ```

5. Store the access token securely for use in your application.

## Quickstart

To use the **HubSpot CRM Object Products** connector in your Ballerina application, update the `.bal` file as follows:

### Step 1: Import the module

Import the `hubspot.crm.obj.products` module and `oauth2` module.

```ballerina
import ballerina/oauth2;

> mport ballerinax/hubspot.crm.obj.products as products;

Reviewer: ```suggestion
mport ballerinax/hubspot.crm.obj.products as hsproducts;
```

Author: Done:[cf70c14](https://github.com/ballerina-platform/module-ballerinax-hubspot.crm.object.products/pull/2/commits/cf70c14111bd6f1e2578ae1728d8b47516dbafec)

```

### Step 2: Instantiate a new connector

1. Create a `Config.toml` file and, configure the obtained credentials in the above steps as follows:

   ```toml
    clientId = <Client Id>
    clientSecret = <Client Secret>
    refreshToken = <Refresh Token>
   ```

> 2. Instantiate a `OAuth2RefreshTokenGrantConfig` with the obtained credentials and initialize the connector with it.
> 
>    ```ballerina
>    configurable string clientId = ?;
>    configurable string clientSecret = ?;
>    configurable string refreshToken = ?;
> 
>    OAuth2RefreshTokenGrantConfig auth = {
>        clientId,
>        clientSecret,
>        refreshToken,
>        credentialBearer: oauth2:POST_BODY_BEARER
>    };
> 
>    products:Client hubspotCrmObjectProducts = check new ({auth: testAuth});

Reviewer: ```suggestion
2. Instantiate a `hsproducts:OAuth2RefreshTokenGrantConfig` with the obtained credentials and initialize the connector with it.

   ```ballerina
   configurable string clientId = ?;
   configurable string clientSecret = ?;
   configurable string refreshToken = ?;

   hsproducts:OAuth2RefreshTokenGrantConfig auth = {
       clientId,
       clientSecret,
       refreshToken,
       credentialBearer: oauth2:POST_BODY_BEARER
   };

   hsproducts:Client hubSpotProducts = check new ({ auth });
```

Author: Done:[cf70c14](https://github.com/ballerina-platform/module-ballerinax-hubspot.crm.object.products/pull/2/commits/cf70c14111bd6f1e2578ae1728d8b47516dbafec)

   ```

### Step 3: Invoke the connector operation

Now, utilize the available connector operations. A sample usecase is shown below.

#### Create a product

```ballerina
public function main() returns error? {

>   products: SimplePublicObjectInputForCreate payload = {
>         "properties": {
>             "name": "Coffee Grinder",
>             "hs_sku": "CMK12346",
>             "price": "540.00",
>             "description": "A high-quality coffee grinder that powders excellent coffee."
>         }
>     };
>     products: SimplePublicObject response = check hubspotCrmObjectProducts->/.post(payload);

Reviewer: ```suggestion
  hsproducts:SimplePublicObjectInputForCreate payload = {
        "properties": {
            "name": "Coffee Grinder",
            "hs_sku": "CMK12346",
            "price": "540.00",
            "description": "A high-quality coffee grinder that powders excellent coffee."
        }
    };
    hsproducts:SimplePublicObject response = check hubSpotProducts->/.post(payload);
```

Author: Done: [cf70c14](https://github.com/ballerina-platform/module-ballerinax-hubspot.crm.object.products/pull/2/commits/cf70c14111bd6f1e2578ae1728d8b47516dbafec)

Reviewer: This is also not done. Can you check ?

Author: Done: [4f527aa](https://github.com/ballerina-platform/module-ballerinax-hubspot.crm.object.products/pull/2/commits/4f527aac3e493b817d3af3c2d5fcc69bc17b1b9e)

> [//]: # (TODO: Add overview mentioning the purpose of the module, supported REST API versions, and other high-level details.)

Reviewer: ```suggestion
```

Author: Done

[HubSpot](https://www.hubspot.com/our-story) is an AI-powered customer relationship management (CRM) platform.  

The `ballerinax/hubspot.crm.obj.products` package offers APIs to connect and interact with [HubSpot API for CRM Object Products](https://developers.hubspot.com/docs/reference/api/crm/objects/products) endpoints, specifically based on [HubSpot CRM Object Products API v3 OpenAPI spec](https://github.com/HubSpot/HubSpot-public-api-spec-collection/blob/main/PublicApiSpecs/CRM/Products/Rollouts/424/v3/products.json).

Use the products API to create and manage your product library, which is the collection of goods and services that your company offers. Products in your product library can be used as line items when creating deals, payment links, invoices, and quotes.

## Setup guide

> [//]: # (TODO: Add detailed steps to obtain credentials and configure the module.)

Reviewer: ```suggestion
```

Author: Done

The `Ballerina HubSpot CRM Products Connector` connector provides practical examples illustrating usage in various scenarios. Explore these [examples](https://github.com/module-ballerinax-hubspot.crm.object.products/tree/main/examples/), covering the following use cases:

> [//]: # (TODO: Add examexport packageUser=ples)

Reviewer: ```suggestion
```

Author: Done

## Useful links

> - For more information go to the [`hubspot.crm.object.products` package](https://central.ballerina.io/ballerinax/hubspot.crm.object.products/latest).

Reviewer: ```suggestion
- For more information go to the [`hubspot.crm.obj.products` package](https://central.ballerina.io/ballerinax/hubspot.crm.obj.products/latest).
```

Author: Should I change 'object' to 'obj' in the image URLs as well?

Author: Done:[cf70c14](https://github.com/ballerina-platform/module-ballerinax-hubspot.crm.object.products/pull/2/commits/cf70c14111bd6f1e2578ae1728d8b47516dbafec)
